tRecorderApi/api/models/take.py

```diff
@@ -33,3 +48,141 @@ def get_takes(chunk_id):
+
+            # Create mode in database if it does not exist
+            mode_obj, m_created = Mode.objects.get_or_create(  # TODO check with joe unique constraint
+                name=manifest["mode"]['name'],
+                slug=manifest["mode"]['slug'],
+                defaults={
+                    'slug': manifest["mode"]["slug"],
+                    'name': manifest["mode"]["name"]
+                }
+            )
+
+            # Create Project in database if it's not there
+            project_obj, p_created = Project.objects.get_or_create(
+                version=version_obj,
+                mode=mode_obj,
+                anthology=anthology_obj,
+                language=language_obj,
+                book=book_obj,
+                published=published,
+                defaults={
+                    'version': version_obj,
+                    'mode': mode_obj,
+                    'anthology': anthology_obj,
+                    'language': language_obj,
+                    'book': book_obj,
+                    'published': published,
+                    'source_language': language_obj  # TODO create source language
+                },
+            )
+
+            manifest_chapter = int(meta['chapter']) - 1
+            checked_level = manifest['manifest'][manifest_chapter]["checking_level"]
+
+            # Create Chapter in database if it's not there
+            chapter_obj, cr_created = Chapter.objects.get_or_create(
+                project=project_obj,
+                number=meta['chapter'],
+                defaults={
+                    'number': meta['chapter'],
+                    'checked_level': checked_level,
+                    'project': project_obj},
+            )
+
+            # Create Chunk in database if it's not there
+            chunk_obj, ck_created = Chunk.objects.get_or_create(
+                chapter=chapter_obj,
+                startv=meta['startv'],
+                endv=meta['endv'],
+                defaults={
+                    'startv': meta['startv'],
+                    'endv': meta['endv'],
+                    'chapter': chapter_obj},
+            )
+
+            markers = json.dumps(meta['markers'])
+
+            # If the take came from .tr file (Source audio)
+            # then check if it exists in database
+            # if it exists then update its data
+            # otherwise create new record
+            # TODO remove source files functionality
+            if published:
+                defaults = {
+                    'location': relpath,
+                    'duration': take_data['duration'],
+                    'rating': 0,  # TODO get rating from tR
+                    'markers': markers,
+                }
+                try:
+                    obj = Take.objects.get(
+                        chunk=chunk_obj,
+                    )
+                    if os.path.exists(obj.location):
+                        os.remove(obj.location)

```

Comment:
this should be done as a method call of a somewhere else, the FileUtility class is an okay start, where we can pass the OS methods off to, but eventually we need to handle this as an interface and the subclass should handle the implementation.

Basically, Django wants to remove any take "artifacts" which, in this case is an audio file. Django should just be responsible for saying hey, I want these files gone, and then something who's responsibility it is for managing file creation, deletion, storage, etc should handle that work.